### PR TITLE
Revert test change - new file handle open on downloadfile

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
This is fixed by https://github.com/Azure/azure-sdk-for-go/pull/22357. 
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

**Comment by tanyasethi-msft** [comment]
To be merged after azblob version update.to 1.3.0

File: sdk/storage/azdatalake/file/client_test.go [hunk 1]
```diff
@@ -4237,18 +4231,8 @@ func (s *RecordedTestSuite) TestFileUploadDownloadSmallFile() {
 	_require.NoError(err)
 	_require.Equal(cnt, fileSize)
 
-	err = destFile.Close()
```
Comment (souravgupta-msft) [inline sdk/storage/azdatalake/file/client_test.go:4240]: Add a defer method for closing the file handle in `TestFileUploadDownloadSmallFile` and `TestFileUploadDownloadSmallFileWithRange`.

Comment (tanyasethi-msft) [reply sdk/storage/azdatalake/file/client_test.go:4240]: missed this. Thanks! 
Added.

File: sdk/storage/azdatalake/go.mod [hunk 2]
```diff
@@ -3,25 +3,25 @@ module github.com/Azure/azure-sdk-for-go/sdk/storage/azdatalake
...
 
 require (
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.9.1
-	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.4.0
-	github.com/Azure/azure-sdk-for-go/sdk/internal v1.5.1
-	github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v1.3.0-beta.1
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.9.2
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.5.1
+	github.com/Azure/azure-sdk-for-go/sdk/internal v1.5.2
+	github.com/Azure/azure-sdk-for-go/sdk/storage/azblob v1.3.0
```
Comment (souravgupta-msft) [inline sdk/storage/azdatalake/go.mod:9]: Please update the changelog for these dependencies

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**